src/sagemaker/model.py

```diff
@@ -437,7 +447,12 @@ def deploy(
         )
 
         if self.predictor_cls:
-            return self.predictor_cls(self.endpoint_name, self.sagemaker_session)
+            predictor = self.predictor_cls(self.endpoint_name, self.sagemaker_session)
+            if serializer:
+                predictor.serializer = serializer
+            if deserializer:
+                predictor.deserializer = deserializer
+            return predictor
```

Comment:
out of scope for this PR (and can be after the v2 release), but perhaps it would make sense to change all the predictor classes to accept optional serializer/deserializer overrides